### PR TITLE
Fix click on link text when other user's cursor is on it.

### LINKED ISSUE #72: Other user's cursor position prevents clicking on link text.
This is similar to [this](https://github.com/reedsy/quill-cursors/issues/32) issue. The other user's cursor position prevents clicking on the link text via the cursor.

### PR DESCRIPTION
**Pull request by vivek-eraser**
Fixes #72

### CODE REVIEW

File: assets/quill-cursors.scss [hunk 1]
```diff
@@ -106,6 +106,7 @@ $transition-func: cubic-bezier(0.250, 0.460, 0.450, 0.940);
   .ql-cursor-caret-container {
     margin-left: -$spacing-sm;
     padding: 0 $spacing-sm;
+    pointer-events: none;
```
[the remark this answers is not in the document]
Comment (vivek-eraser) [reply assets/quill-cursors.scss:109]: Other alternative can be tweaking the z-index but that feels a bit hacky.

[the remark this answers is not in the document]
Comment (vivek-eraser) [reply assets/quill-cursors.scss]: Agreeing that it breaks existing UX, I have pushed another change which reduces padding width to allow more room for cursor clicks.

File: assets/quill-cursors.scss [hunk 2]
```diff
@@ -104,9 +104,6 @@ $transition-func: cubic-bezier(0.250, 0.460, 0.450, 0.940);
   }
 
   .ql-cursor-caret-container {
-    margin-left: -$spacing-sm;
-    padding: 0 $spacing-sm;
```
Comment (alecgibson) [inline assets/quill-cursors.scss:108]: This is here to make it easier to trigger the hover behaviour; I'm afraid we can't just delete it.

**Comment by alecgibson** [comment]
Okay, I've had a little think about this and the best CSS-only solution I can come up with is this:

```scss
  .ql-cursor-caret-container {
    // ...existing styles

    // Avoid flicker when applying pointer-events:none on hover
    cursor: text;

    &:hover {
      pointer-events: none;
    }
  }
```

This _mostly_ works, with the major exception of mobile devices, which obviously don't trigger the `:hover` selector.

I'm starting to think that this issue is sufficiently complex that we should move the flag display toggling from CSS to JS, where we might have a bit more control, but I don't really have time to think about it or implement it right now.

If my suggested CSS-only solution is "good enough" for you, then I'm happy to merge that as an improvement on current behaviour, even if it doesn't 100% solve the issue, and hopefully I'll get around to looking at mobile devices at some point when I have time.

**Comment by vivek-eraser** [comment]
I am fine with this fix as far as it works(agreeing that we don't want this for the mobile). In the long term it would be better to control the Flag behaviour via JS(I will plan to work on this in next couple of weeks).

Thanks for looking into this.

**Review by alecgibson — APPROVED** [review]

**Pull request merged**